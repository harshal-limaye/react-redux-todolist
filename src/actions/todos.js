export const addToDo = text => ({ type: 'ADD', text });

export const completeToDO = id => ({ type: 'COMPLETE', id });

export const deleteToDO = id => ({ type: 'DELETE', id });