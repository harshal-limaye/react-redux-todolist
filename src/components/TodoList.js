import React from 'react';
import { connect } from 'react-redux';
import Todo from './Todo';

class TodoItems extends React.Component {

    render() {
        return (
            this.props.todos && this.props.todos.length > 0 ?
            <div className="todo-items">
                <ul>
                    {
                        this.props.todos.map((todo) => <Todo key={todo.id} todo={todo} />)
                    }
                </ul>
            </div>
            : false
        );
    }
}

const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps)(TodoItems);