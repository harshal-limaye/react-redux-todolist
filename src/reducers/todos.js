const todos = (state = [], action) => {
    switch (action.type) {
        case 'ADD':
            return [ { id: state.length, text: action.text, completed: false }, ...state ];
        case 'COMPLETE':
            return state.filter(t => t.id === action.id ? Object.assign({}, t, {completed: !t.completed}) : t);
        case 'DELETE':
            return state.filter(t => t.id !== action.id);
        default:
            return state;
    }
}

export default todos;