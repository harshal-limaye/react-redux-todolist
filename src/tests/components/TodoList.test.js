import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../../components/TodoList';
import todos from '../fixtures/todos';

describe('testing todolist component', () => {
    test('should render todolist component with todos', () => {
        const wrapper = shallow(<TodoList todos={todos} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render todolist component without todos', () => {
        const wrapper = shallow(<TodoList todos={[]} />);
        expect(wrapper).toMatchSnapshot();
    });
})