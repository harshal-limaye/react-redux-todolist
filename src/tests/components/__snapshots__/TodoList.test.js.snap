// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing todolist component should render todolist component with todos 1`] = `
<div
  className="todo-items"
>
  <ul>
    <Todo
      key="3"
      todo={
        Object {
          "completed": true,
          "id": 3,
          "text": "Write test cases",
        }
      }
    />
    <Todo
      key="2"
      todo={
        Object {
          "completed": false,
          "id": 2,
          "text": "create container components",
        }
      }
    />
    <Todo
      key="1"
      todo={
        Object {
          "completed": false,
          "id": 1,
          "text": "fix bugs",
        }
      }
    />
    <Todo
      key="0"
      todo={
        Object {
          "completed": false,
          "id": 0,
          "text": "release next version",
        }
      }
    />
  </ul>
</div>
`;

exports[`testing todolist component should render todolist component without todos 1`] = `
<div
  className="todo-items"
>
  <ul />
</div>
`;
